### PR TITLE
2.0 526 misc fixes 1

### PR DESCRIPTION
**Pull request by whitespine**
- Fixed NPC feature acc/bonus clickers.
- Fixed adding npc features to classes/templates
- Fixed weapon mods type/size selectors
- Added capability to mandate a minimum/specific size to some arrays (e.g. npc stats, npc feature acc/bonus)
- Weapons now have minnimum one profile
- LID lookups can hit world documents
- LID lookups now have sync capability
  - Still haven't dealt with combining mine / bolts

### CODE REVIEW

**Comment by whitespine** [comment]
ALSO: Fixed deployables name on import
Fixed license sheets

File: src/lancer.ts [hunk 1]
```diff
@@ -555,13 +557,13 @@ Hooks.once("init", async function () {
   Handlebars.registerHelper("item-edit-arrayed-damage", item_edit_arrayed_damage);
   Handlebars.registerHelper("item-edit-arrayed-range", item_edit_arrayed_range);
   Handlebars.registerHelper("item-edit-arrayed-enum", item_edit_arrayed_enum);
+  Handlebars.registerHelper("item-edit-checkboxes-object", item_edit_checkboxes_object);
   Handlebars.registerHelper("item-edit-arrayed-bonuses", item_edit_arrayed_bonuses);
   Handlebars.registerHelper("item-edit-arrayed-counters", item_edit_arrayed_counters);
   Handlebars.registerHelper("item-edit-arrayed-deployables", item_edit_arrayed_deployables);
   Handlebars.registerHelper("item-edit-arrayed-synergies", item_edit_arrayed_synergies);
   Handlebars.registerHelper("item-edit-arrayed-integrated", item_edit_arrayed_integrated);
   // Generic handler for an array that can take a selectable enum
```
Comment (Eranziel) [inline src/lancer.ts:566]: Orphaned comment.

File: src/module/item/license-sheet.ts [hunk 2]
```diff
@@ -32,13 +33,22 @@ export class LancerLicenseSheet extends LancerItemSheet<EntryType.LICENSE> {
...
       if (pack) {
-        let docs = (await pack.getDocuments({ system: { license: license.system.key } })) as any;
-        for (let d of docs as LancerItem[]) {
-          let rank = (d as any).system.license_level as number;
+        let index = await pack.getIndex();
+        // @ts-expect-error
+        let key = this.item.system.key;
+        for (let [id, index_data] of index.entries()) {
+          // @ts-expect-error
+          let item_license = index_data.system.license as string | undefined;
```
Comment (Eranziel) [inline src/module/item/license-sheet.ts:41]: I added a typedef for `LancerActor.system` and `LancerItem.system`, so these probably won't throw errors anymore.

File: src/module/util/lid.ts [hunk 3]
```diff
@@ -53,14 +52,22 @@ const lookupLIDPluralCache = new RepentantFetcherCache<string, Array<LancerActor
       }
     }
 
+    // Also dig through world items
+    for (let item of game.items!.contents as LancerItem[]) {
+      // @ts-expect-error
+      if (item.system.lid == lid && raw_types.includes(item.type)) {
```
Comment (Eranziel) [inline src/module/util/lid.ts:58]: TS error should be gone now.

**Review by Eranziel — APPROVED** [review]
A couple notes for cleanup after merge.

File: src/module/item/license-sheet.ts (hunk 2 above)
Comment (Eranziel) [reply src/module/item/license-sheet.ts:41]: Never mind, it's not complaining about `LancerItem.system` here.

Comment (whitespine) [reply src/module/item/license-sheet.ts:41]: Oh, thanks. I tried but just got an infinitely recursive type error, lol.

**Pull request merged**